File: svws-webclient/core/src/asd/validate/schueler/ValidatorSbe01SchuelerSchulbesuchsdatenEinschulungsart.ts
import { ValidatorSbe02SchuelerSchulbesuchsdatenEinschulungsart } from '../../../asd/validate/schueler/ValidatorSbe02SchuelerSchulbesuchsdatenEinschulungsart';
import { SchuelerSchulbesuchsdaten } from '../../../asd/data/schueler/SchuelerSchulbesuchsdaten';
import { Einschulungsart } from '../../../asd/types/schueler/Einschulungsart';
import type { Supplier } from '../../../java/util/function/Supplier';
import { Class } from '../../../java/lang/Class';
import { ValidatorKontext } from '../../../asd/validate/ValidatorKontext';
import { Validator } from '../../../asd/validate/Validator';

export class ValidatorSbe01SchuelerSchulbesuchsdatenEinschulungsart extends Validator {

	private readonly _schulbesuchsdaten: Supplier<SchuelerSchulbesuchsdaten>;


	/**
	 * Erstellt einen neuen Validator mit den übergebenen Daten und dem Kontext.
	 *
	 * @param schulbesuchsdaten   ein Supplier für die Schulbesuchsdaten des Schülers
	 * @param kontext             der Kontext des Validators
	 */
	public constructor(schulbesuchsdaten: Supplier<SchuelerSchulbesuchsdaten>, kontext: ValidatorKontext) {
		super(kontext);
		this._schulbesuchsdaten = schulbesuchsdaten;
		this._validatoren.add(new ValidatorSbe02SchuelerSchulbesuchsdatenEinschulungsart(schulbesuchsdaten, kontext));
	}

	protected pruefe(): boolean {
		const daten: SchuelerSchulbesuchsdaten | null = this._schulbesuchsdaten.get();
		if ((daten === null) || (daten.idEinschulungsartGrundschule === null) || (daten.idEinschulungsartGrundschule < 0)) {
			return true;
		}
		if (Einschulungsart.data().getWertByIDOrNull(daten.idEinschulungsartGrundschule) === null) {
			this.addFehler(0, "Einschulungsart des Schülers: Das Feld 'Einschulungsart' muss zulässig sein.");
			return false;
		}
		return true;
	}

	transpilerCanonicalName(): string {
		return 'de.svws_nrw.asd.validate.schueler.ValidatorSbe01SchuelerSchulbesuchsdatenEinschulungsart';
	}

	isTranspiledInstanceOf(name: string): boolean {
		return ['de.svws_nrw.asd.validate.BasicValidator', 'de.svws_nrw.asd.validate.schueler.ValidatorSbe01SchuelerSchulbesuchsdatenEinschulungsart', 'de.svws_nrw.asd.validate.Validator'].includes(name);
	}

	public static readonly class = new Class<ValidatorSbe01SchuelerSchulbesuchsdatenEinschulungsart>('de.svws_nrw.asd.validate.schueler.ValidatorSbe01SchuelerSchulbesuchsdatenEinschulungsart');

}

export function cast_de_svws_nrw_asd_validate_schueler_ValidatorSbe01SchuelerSchulbesuchsdatenEinschulungsart(obj: unknown): ValidatorSbe01SchuelerSchulbesuchsdatenEinschulungsart {
	return obj as ValidatorSbe01SchuelerSchulbesuchsdatenEinschulungsart;
}

File: svws-asd/src/main/java/de/svws_nrw/asd/validate/schueler/ValidatorSbe01SchuelerSchulbesuchsdatenEinschulungsart.java
package de.svws_nrw.asd.validate.schueler;

import java.util.function.Supplier;

import de.svws_nrw.asd.data.schueler.SchuelerSchulbesuchsdaten;
import de.svws_nrw.asd.types.schueler.Einschulungsart;
import de.svws_nrw.asd.validate.Validator;
import de.svws_nrw.asd.validate.ValidatorKontext;
import jakarta.validation.constraints.NotNull;

/**
 * Validator SBE01: Prüft, ob die angegebene Einschulungsart ein gültiger
 * und zulässiger Eintrag im Katalog ist.
 */
public final class ValidatorSbe01SchuelerSchulbesuchsdatenEinschulungsart extends Validator {

	private final @NotNull Supplier<SchuelerSchulbesuchsdaten> _schulbesuchsdaten;

	/**
	 * Erstellt einen neuen Validator mit den übergebenen Daten und dem Kontext.
	 *
	 * @param schulbesuchsdaten   ein Supplier für die Schulbesuchsdaten des Schülers
	 * @param kontext             der Kontext des Validators
	 */
	public ValidatorSbe01SchuelerSchulbesuchsdatenEinschulungsart(
			final @NotNull Supplier<SchuelerSchulbesuchsdaten> schulbesuchsdaten,
			final @NotNull ValidatorKontext kontext) {
		super(kontext);
		_schulbesuchsdaten = schulbesuchsdaten;
		_validatoren.add(new ValidatorSbe02SchuelerSchulbesuchsdatenEinschulungsart(schulbesuchsdaten, kontext));
	}

	@Override
	protected boolean pruefe() {
		final SchuelerSchulbesuchsdaten daten = _schulbesuchsdaten.get();

		if ((daten == null) || (daten.idEinschulungsartGrundschule == null) || (daten.idEinschulungsartGrundschule < 0)) {
			return true;
		}

		if (Einschulungsart.data().getWertByIDOrNull(daten.idEinschulungsartGrundschule) == null) {
			addFehler(0, "Einschulungsart des Schülers: Das Feld 'Einschulungsart' muss zulässig sein.");
			return false;
		}

		return true;
	}

}
